File: LeetCode/March/Delete_and_earn.java
class Solution {
    public int deleteAndEarn(int[] nums) {
        int[] count = new int[10001];
            for (int num : nums) {
                count[num] += num;
            }
        int pick = 0; 
        int notPick =count[1];
        
        for(int i =2; i <= 10000;i++){
            int curr = Math.max(pick + count[i], notPick);
            pick = notPick;
            notPick = curr;
        }
        
        return notPick;
    }
}

File: LeetCode/March/Delete_and_earn.cpp
class Solution {
public:
    int deleteAndEarn(vector<int>& nums) {
        vector<int> count(10005, 0);
        
        for(int x : nums){
            count[x]+=x;
        }
        
        int pick = 0;
        int notPick = 0;
        for(int i = 1; i < 10005; i++){
            int curr = max(count[i]+pick, notPick);
            pick = notPick;
            notPick = curr;
        }
        return notPick;
    }
};